### PR TITLE
Tracking when a user logs out or is logged out

### LINKED ISSUE #2646: Log on MixPanel Logout events
Log explicit logout events done by the user and also events that happen because we detected incorrect states on the API calls ( wrong or expired tokens).

This may help to find issues where users can't login or get logged out in the app.

### PR DESCRIPTION
**Pull request by sendhil**
Fixes #2646

cc @diegoreymendez

### CODE REVIEW

File: WordPress/Classes/System/WordPressAppDelegate.m [hunk 1]
```diff
@@ -1239,6 +1239,9 @@ - (void)handleDefaultAccountChangedNotification:(NSNotification *)notification
         }];
     } else {
         // No need to check for welcome screen unless we are signing out
```
Comment (diegoreymendez) [inline WordPress/Classes/System/WordPressAppDelegate.m]: I know this is not part of your PR, but do you know what this comment means?  It sounds to me like it's either outdated or doesn't have enough details to be of value.  Thoughts?

File: WordPress/Classes/ViewRelated/Blog/BlogListViewController.m [hunk 2]
```diff
@@ -319,6 +319,7 @@ - (void)tableView:(UITableView *)tableView
                 WPAccount *defaultAccount = [accountService defaultWordPressComAccount];
 
                 if (!defaultAccount) {
+                    [WPAnalytics track:WPAnalyticsStatLogout];
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Blog/BlogListViewController.m]: Code looks ok, but I find it that we're logging the user out in weird places in our code.  This is one of those cases.

**Comment by sendhil** [comment]
@diegoreymendez  - pushed some fixes.

**Comment by diegoreymendez** [comment]
:shipit:

**Pull request merged**